lib/importproject.cpp

```diff
@@ -108,18 +108,34 @@ static bool simplifyPathWithVariables(std::string &s, const std::map<std::string
         if (expanded.find(var) != expanded.end())
             break;
         expanded.insert(var);
-        std::map<std::string, std::string>::const_iterator it1 = variables.find(var);
-        if (it1 == variables.end())
-            break;
-        s = s.substr(0,start) + it1->second + s.substr(end+1);
+		std::map<std::string, std::string>::const_iterator it1 = variables.find(var);
+		std::string variableValue;
+		// variable was not found within defined variables
+		if (it1 != variables.end()) {
+			variableValue = it1->second;
+		}
+		// also search environment variables
+		else {
+			char const* envValue = std::getenv(var.c_str());
+			if (envValue) {
+				variableValue = std::string(envValue);
+				// also cache value for next time to reduce system access
+				variables[var] = variableValue;
+			}
+			else {
+				break;

```

Comment:
i implemented this but took it out again in [this commit ](https://github.com/danmar/cppcheck/pull/920/commits/c09227a454088d9394c5fba61b3da195e5bb2f36). If its ok with you, we can put it back in. It wont overwrite "projectDir" and so on, since they were defined first.